File: Data Structures & Algorithms/validate-parentheses/submission-0.js
class Solution {
    /**
     * @param {string} s
     * @return {boolean}
     */
    isValid(s) {
       let map = {
        "]" : "[", 
        ")" : "(", 
        "}" : "{"
       }
       let stack = []; 
       for (let char of s) { 
        if (char in map ){
            if ( stack.length === 0 || stack.pop() !== map[char]){
                return false ; 
            }
        }else { 
            stack.push(char)
        }
       }
       return stack.length === 0 ; 
    }
}

File: Data Structures & Algorithms/largest-rectangle-in-histogram/submission-0.py
class Solution:
    def largestRectangleArea(self, heights: List[int]) -> int:
        heights.append(0)
        stack = []
        maxArea = 0 

        for i in range(len(heights)) : 
            while stack and heights[stack[-1]] > heights[i] : 
                height = heights[stack.pop()]
                left = stack[-1] if stack else -1 
                right = i 
                width = right - left -1 
                area = height * width 
                maxArea = max(maxArea , area)
            stack.append(i)
        return maxArea